### ISSUE TITLE
Failure to build for GPUs

### ISSUE THREAD

**Issue by mpearmain** [dmlc/mxnet#578]
I've pulled the config.mk file and try and build but get the following error

/usr/bin/ld: cannot find -lcudart
/usr/bin/ld: cannot find -lcublas
/usr/bin/ld: cannot find -lcurand
collect2: error: ld returned 1 exit status
make: *** [lib/libmxnet.so] Error 1
make: *** Waiting for unfinished jobs....
/usr/bin/ld: cannot find -lcudart
/usr/bin/ld: cannot find -lcublas
/usr/bin/ld: cannot find -lcurand
collect2: error: ld returned 1 exit status
make: *** [bin/im2rec] Error 1

my ~/.bashrc contains the following (works with GPU for Theano)

 # CUDA
export CUDA_HOME=/usr/local/cuda-7.0 
export CUDA_ROOT=/usr/local/cuda-7.0/bin 
export LD_LIBRARY_PATH=${CUDA_HOME}/lib64 
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda
 
PATH=${CUDA_HOME}/bin:${PATH} 
export PATH

Any suggestions?

**Comment by shuokay** [comment]
Need to set ```USE_CUDA_PATH = /usr/local/cuda-7.0``` in config.mk
https://github.com/dmlc/mxnet/blob/master/make/config.mk#L46

**Comment by mpearmain** [comment]
Path now look like:
>  #---------------------------------------------
>  # matrix computation libraries for CPU/GPU
>  #---------------------------------------------

>  # whether use CUDA during compile
> USE_CUDA = 1

>  # add the path to CUDA libary to link and compile flag
>  # if you have already add them to enviroment variable, leave it as NONE
>  # USE_CUDA_PATH = /usr/local/cuda
> USE_CUDA_PATH = /usr/local/cuda-7.0

Rebuilt R package as per instruction and testing in R.
> mx.gpu()
  $device
  [1] "gpu"
  $device_id
  [1] 0
  $device_typeid
  [1] 2
  attr(,"class")
  [1] "MXContext"

> model <- mx.mlp(train.x, 
+                 train.y, 
+                 hidden_node=10, 
+                 out_node=2, 
+                 out_activation="softmax",
+                 num.round=20, 
+                 array.batch.size=15, 
+                 learning.rate=0.07, 
+                 momentum=0.9, 
+                 eval.metric=mx.metric.accuracy,
+                 device = mx.gpu())

Auto detect layout of input matrix, use rowmajor..

Start training with 1 devices

[09:57:19] ./dmlc-core/include/dmlc/././logging.h:208: [09:57:19] src/storage/storage.cc:43: Please compile with CUDA enabled
Error: [09:57:19] src/storage/storage.cc:43: Please compile with CUDA enabled

**Comment by shuokay** [comment]
You need to rebuild mxnet and R package, because R package calls mxnet C++ library.

**Comment by tqchen** [comment]
closing for now, you will need to compile mxnet with GPU before install R pkg

**Comment by long-jian** [comment]
Hi, I got the same problem  with @mpearmain . My OS is windows10, GPU card is NVIDIA GeForce GTX 1080 with CUDA 7.5 driver. My R is 3.3.0 version.
How can I compile  mxnet with GPU?

**Comment by hjk41** [comment]
@long-jian Did you follow the installation instructions here: http://mxnet.readthedocs.io/en/latest/how_to/build.html#building-on-windows? What was the error message?

**Comment by long-jian** [comment]
@hjk41 The warning is:

>Start training with 1 devices
 [15:08:46] D:\chhong\mxnet\dmlc-core\include\dmlc/logging.h:235: [15:08:46] D:\chhong\mxnet\src\storage\storage.cc:44: Please compile with CUDA enabled
> Error: [15:08:46] D:\chhong\mxnet\src\storage\storage.cc:44: Please compile with CUDA enabled 

and I try to compile mxnet GPU pkg, I follow this page: [(https://github.com/dmlc/mxnet/issues/1085)]
but it's always failure. Is there any easy way to compile a R package?

**Comment by thirdwing** [comment]
You can use the pre-built GPU lib by @hjk41 and follow https://github.com/Lodewic/mxnet/blob/f127a693500f834209aef607ebd21e591334277e/doc/build.md#r-package-installation

**Comment by long-jian** [comment]
Thanks for @hjk41 and @thirdwing , I successfully installed mxnet gpu version on R by following this tutorial, but here is the question, bellowing the third step of explanation "There should be 11 dll files now in R-package/inst/libs/x64", but what I got is 12 files of .dll. here is the list.
  **cublas64_75.dll**
  **cudart64_75.dll**
  **cudnn64_70.dll**
  **curand64_75.dll**
  **libgcc_s_seh-1.dll**
  **libgfortran-3.dll**
  **libmxnet.dll**
  **libopenblas.dll**
  **libquadmath-0.dll**
  **nvrtc64_75.dll**
  **unzip32.dll**
  **vcomp120.dll**
Would it cause any problems of mxnet? Because while I tried MNIST example, the cpu environment works fine, but the gpu environment is always low accuracy about 10%.

**Comment by hjk41** [comment]
Right, there was a PR which makes unzip unnecessary.

On Wed, Jun 22, 2016 at 10:59 AM, Qiang Kou (KK) <notifications@github.com>
wrote:

> unzip might not be necessary.
> On Jun 21, 2016 7:58 PM, "long-jian" <notifications@github.com> wrote:
>
> > Thanks for @hjk41 <https://github.com/hjk41> and @thirdwing
> > <https://github.com/thirdwing> , I successfully installed mxnet gpu
> > version on R by following this tutorial, but here is the question,
> > bellowing the third step of explanation "There should be 11 dll files now
> > in R-package/inst/libs/x64", but what I got is 12 files of .dll. here is
> > the list.
> > *cublas64_75.dll*
> > *cudart64_75.dll*
> > *cudnn64_70.dll*
> > *curand64_75.dll*
> > *libgcc_s_seh-1.dll*
> > *libgfortran-3.dll*
> > *libmxnet.dll*
> > *libopenblas.dll*
> > *libquadmath-0.dll*
> > *nvrtc64_75.dll*
> > *unzip32.dll*
> > *vcomp120.dll*
> > Would it cause any problems of mxnet? Because while I tried MNIST
> example,
> > the cpu environment works fine, but the gpu environment is always low
> > accuracy about 10%.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/dmlc/mxnet/issues/578#issuecomment-227630252>, or
> mute
> > the thread
> > <
> https://github.com/notifications/unsubscribe/ABebVdkaJqcCaQuv_MdmcMAMKlupEmosks5qOKTbgaJpZM4GiUcw
> >
>
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmlc/mxnet/issues/578#issuecomment-227630401>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/ABFI4W2xptowlNvx3M-UeLE9vl2FNGcKks5qOKUtgaJpZM4GiUcw>
> .
>



-- 
HONG Chuntao
System Research Group
Microsoft Research Asia

**Comment by long-jian** [comment]
I'm so appreciated for your helping, I think I successfully installed mxnet on R.

Now I am facing the accuracy problem, which I have same problem like this issue https://github.com/dmlc/mxnet/issues/1228
I will post my question below this page.
